app/src/main/java/org/zephyrsoft/trackworktime/ReportsActivity.java

```diff
@@ -97,132 +92,157 @@ public void onCreate(Bundle savedInstanceState) {
 			unitYear.setEnabled(!isChecked);
 		});
 
-		allEventsButton.setOnClickListener(v -> {
-            Range selectedRange = getSelectedRange();
-            Unit selectedUnit = getSelectedUnit();
+		exportButton.setOnClickListener(v -> export());
+	}
 
-            DateTime[] beginAndEnd = timeCalculator.calculateBeginAndEnd(selectedRange, selectedUnit);
-            List<Event> events = dao.getEvents(beginAndEnd[0], beginAndEnd[1]);
+	private void export() {
+		int selectedId = groupingRadioGroup.getCheckedRadioButtonId();
+		switch (selectedId) {
+			case R.id.groupingNone:
+				exportAllEvents();
+				break;
+			case R.id.groupingByTask:
+				exportTimesByTask();
+				break;
+			case R.id.groupingByTaskPerDay:
+				exportTimesByTaskPerDay();
+				break;
+			case R.id.groupingByTaskPerWeek:
+				exportTimesByTaskPerWeek();
+				break;
+			case R.id.groupingByTaskPerMonth:
+				exportTimesByTaskPerMonth();
+				break;
+			default:
+				throw new RuntimeException("Grouping not implemented");
+		}
+	}
 
-            String report = csvGenerator.createEventCsv(events);
-			String reportName = getNameForSelection(selectedRange, selectedUnit);
-			if (report == null) {
-				logAndShowError("could not generate report " + reportName);
-				return;
-			}
+	private void exportAllEvents() {
```

Comment:
Note: Logic behind the exports stayed exactly the same. I just had to wrap them in methods and indent them.